labs/a10.md

```diff
@@ -0,0 +1,371 @@
+
+The `-i` flag tells docker to keep `STDIN` open to your container, and the `-t` flag allocates a
+[pseudo-TTY](https://en.wikipedia.org/wiki/Pseudoterminal) for you. Basically you need both for you to have a way to
+enter text and have this display properly. At the end of the command, `/bin/bash` is just the command you want to run
+once the container starts up. Try installing some packages from `apt` or just play around. It should look like a bare
+Linux system.
+
+You can exit the container with `CTRL+D`.
+
+Notice how even though your VM is running Debian, you were able to run the Xenial version of Ubuntu in a container. If
+you are curious about other variants of Linux, you can run a lot of them inside containers as well! This all works because
+Linuxdistributions all share the Linux kernel. For that same reason, you won't be able to run MacOS or Windows in a container.
+You can try running [Fedora](https://getfedora.org/) (_\*tips hat\*_ M'Linux), another long-running Linux distribution.:
+
+`docker run -it fedora:latest /bin/bash`
+
+#### Questions (answer on Google form)
+
+1. What user are you logged in as by default?
+2. If you start and then exit an interactive container, and then use the `docker run -it ubuntu:xenial /bin/bash`
+command again, is it the same container? How can you tell?
+
+### Basic Management
+The Docker CLI (Command Line Interface) has some basic commands for you to monitor running and stopped containers,
+downloaded images, and other information. We'll go over the basic commands you'll probably use, but be sure to check
+out [the full reference](https://docs.docker.com/engine/reference/commandline/cli/) if you're interested.
+
+Firstly, you might want to see the running containers on a system. Use the following command:
+
+`docker ps`
+
+Since you (likely) have no containers running, you probably won't see anything interesting. However, if you pass in the
+`-a` flag, you'll also be able to see containers that have stopped (make your terminal wider or it'll display weird):
+
+```
+baisang@rapture ~/d/labs> docker ps -a
+CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS                            PORTS               NAMES
+35c048c03588        fedora:latest       "/bin/bash"              7 minutes ago       Exited (130) About a minute ago                       mystifying_edison
+dd8f7cc2e0cd        fedora:latest       "/bin/bash"              10 minutes ago      Exited (1) 8 minutes ago                              romantic_mahavira
+```
+
+This lets you see a lot of useful information about the container. Observe that each container has a unique container
+id and a unique human-readable name. To get more information about a container, you can use the `docker logs` command
+to fetch the logs of a container (whether it's still running or exited):
+
+`docker logs <container_id_or_name>`
+
+This basically just gives you `stdout` and `stderr` for the container.
```

Comment:
I'd probably say the "process(es) running in the container" instead of the container itself to make it clear where the `stdout` and `stderr` are actually generated.